src/core/data/particles/particle_array.hpp

```diff
@@ -230,6 +230,22 @@ class ParticleArray
     auto& vector() { return particles_; }
     auto& vector() const { return particles_; }
 
+    auto& update_from(This const& that)
+    {
+        bool resize_bigger = this->size() < this->size();
+        if (resize_bigger)
+            this->resize(this->size() + static_cast<std::size_t>(this->size() * .05));
+        else
+            this->resize(that.size());
+        std::copy(that.begin(), that.end(), this->begin());
+        this->box_     = that.box_;
+        this->cellMap_ = that.cellMap_;
+        return *this;
+    }
+
+
+    void shrink_to_fit() { particles_.shrink_to_fit(); }
```

Comment:
Mayby wrong reasons but always had some bad feelings about this `shrink_to_fit` function.
Had some issues in the past were it led to unexpected results, can't remember well.
found this : http://slashslash.info/2013/11/capacity-members-for-vector-shrink_to_fit/

Reply:
I commented on this not always working in updater.hpp, alternatively, we can have a unique_ptr<vector> and reset/reallocate it if we want to free up some memory